### DISCUSSION TITLE
Folders step 2: Add multiple assignment and category nesting

### DISCUSSION THREAD

**Discussion by kuzmany** [mautic/mautic#13020] [Announcements]
Based on https://docs.google.com/document/d/1nJxRDGslOg5aIxoLyK2RwGz1zek9i_DD9L555vnDHN4/edit

> Multiple Categories Per Entity: With this feature enhancement, you will be able to assign more than one category to a resource. This enhancement will increase the system's versatility and allow you to better manage crossover resources – like assets which belong to multiple projects or marketing campaigns, for example.

> Category Nesting: This feature will give you the power to nest categories within each other, further enhancing the tool's organization capabilities.

Both of these features will be created on the backend with a simple UI using the current Mautic design system. Additional UI tweaks can be added in the next steps. We would primarily focus on the backend at this stage.

**Comment by RCheesley** [comment]
Pinging @mautic/ux-ui-tiger-team for their feedback/thoughts.

**Reply by kuzmany** [reply]
I prefer to use something that already exists. For example, multiple select like we use in companies assignment in the contact forms:

![image](https://github.com/mautic/mautic/assets/462477/884ef3fa-8f51-4097-af39-d40d4baf4289)

**Reply by kuzmany** [reply]
I think we should stick to a maximum of 3 levels of hierarchy. That should be enough.

**Comment by andersonjeccel** [comment]
@code5rick
